src/algebra/module/submodule_bilinear.lean

```diff
@@ -0,0 +1,129 @@
+/-
+Copyright (c) 2019 Kenny Lau. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kenny Lau, Eric Wieser
+-/
+import linear_algebra.span
+import linear_algebra.bilinear_map
+
+/-!
+# Images of pairs of submodules under bilinear maps
+
+This file provides `submodule.map₂`, which is later used to implement `submodule.has_mul`.
+
+## Main results
+
+* `submodule.map₂_eq_span_image2`: the image of two submodules under a bilinear map is the span of
+  their `set.image`.
```

Comment:
```suggestion
  their `set.image`.

## Notes

This file is very similar to the n-ary section of `data.set.basic` and to `order.filter.n_ary`.
Please keep them in sync.
```
And can you update the similar warnings in `data.set.basic` and `order.filter.n_ary`?

Reply:
Is it actually similar right now?

Reply:
*somehow*, but not fully indeed. That should probably be fixed.
```suggestion
  their `set.image`.
## Notes

This file is quite similar to the n-ary section of `data.set.basic` and to `order.filter.n_ary`.
Please keep them in sync.
```